src/main.rs

```diff
@@ -81,16 +88,78 @@ enum Cmd {
         #[clap(long)]
         leave_debug_open: bool,
         /// Requires that a subset of keys are enabled (and only that subset).
-        /// This flag takes a bitmask (in the base of your choice as long as
-        /// it's not octal) where bit 0 maps to RoTK 0, 1 maps to 1, and so
-        /// forth. If keys corresponding to 1 bits are not enabled, and keys
-        /// corresponding to 0 bits are not *not* enabled, this will decline to
-        /// lock.
-        #[clap(long, value_parser = parse_int::parse::<u8>)]
-        require_key_enable_shape: Option<u8>,
+        /// This flag takes a 4 character string where 'E' represents
+        /// enabled, 'R' revoked, and 'I' invalid. Keys are specified
+        /// in order from 3 to 0. A string of EERR means keys 0 and 1
+        /// are revoked and keys 2 and 3 are enabled.
+        /// If the bits do not match this will decline to lock!
+        #[clap(long, value_parser = parse_key_shape)]
+        require_key_enable_shape: Option<String>,
     },
 }
 
+fn rotkh_revoke_to_ascii(revoke: u32) -> String {
+    let mut chars: [char; 4] = [' '; 4];
+    for (slot, entry) in chars.iter_mut().enumerate() {
+        *entry = match revoke >> (2 * slot) & 0x3 {
+            0b00 => 'i',
+            0b01 => 'e',
+            _ => 'r',
+        }
+    }
+
+    format!("{}{}{}{}", chars[3], chars[2], chars[1], chars[0])
+}
+
+fn parse_key_shape(input: &str) -> Result<String> {
+    if input.len() != 4 {
+        bail!("Must specify exactly 4 settings");
+    }
+
+    for c in input.chars() {
+        match c {
+            'E' | 'e' | 'I' | 'i' | 'R' | 'r' => (),
+            'b' => {
```

Comment:
What's the context here?  Was this previously a valid setting?

Reply:
We used to specify a mask in binary, like `0b0011`

Reply:
Wouldn't that bail out on the first `0`, then, instead of making it to the `b` and printing the helpful message?

Reply:
Yeah I was attempting to give a hint to the user that this had changed but you're right that if we're going to have the message it should be on `0` not `b`